test/sqllogictest/joins.slt

```diff
@@ -602,3 +602,92 @@ CREATE TABLE join_fail (la date);
 
 query error NATURAL/USING join column "la" cannot be cast to uniform type: integer vs date
 SELECT la FROM l JOIN join_fail USING (la)
+
+# test that joins properly handle null keys
+statement ok
+CREATE TABLE l3 (la int, lb text)
+
+statement ok
```

Comment:
In this example, the entire right side of the outer join and the WHERE clause is yanked away:

```
materialize=> EXPLAIN SELECT * FROM t1 LEFT JOIN t2 ON (t1.f2 = t2.f1) WHERE t2.f2 IS NULL;
          Optimized Plan           
-----------------------------------
 %0 =                             +
 | Get materialize.public.t1 (u57)+
 | Map null, null                 +
```
This produces a wrong result .